src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getSearchWith } from '../utils/searchHelper';
+import { getVisiblePeople } from '../helpers/getVisiblePeople';
+import { getPreparedPeople } from '../helpers/getPreparedPeople';
+
+export const PeoplePage: React.FC = React.memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDataLoaded, setIsDataLoaded] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const loadPeople = async () => {
+    setIsLoading(true);
+
+    try {
+      const loadedPeople = await getPeople();
+
+      const preparedPeople = getPreparedPeople(loadedPeople);
+
+      setPeople(preparedPeople);
+      setIsDataLoaded(true);
+    } catch (error) {
+      setErrorMessage('Can\'t load people');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const handleQueryChange = useCallback((
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setSearchParams(
+      getSearchWith(searchParams, { query: event.target.value || null }),
+    );
+  }, [query, searchParams]);
+
+  const visiblePeople = useMemo(() => getVisiblePeople(
+    people,
+    sex,
+    query,
+    centuries,
+    sort,
+    order,
+  ), [people, sex, query, centuries, sort, order]);
```

Comment:
The function that takes 3 or more arguments is hard to maintain. It's really easy to mess up the order of params, so you should pass an object.

Reply:
Also, it's better to create a custom hook that returns the result of useMemo, than create a function and call it inside useMemo hooks.

```
export const useVisiblePeople = (params) => {
  return useMemo(() => {
   // filter logic
}, [bla])
}

Reply:
done